packages/syft/src/syft/service/network/network_service.py

```diff
@@ -228,19 +237,39 @@ def add_peer(
                 message="verify_key does not match the remote node's verify_key for add_peer"
             )
 
+        # check if the peer already is a node peer
+        existed_peer = self.stash.get_by_uid(context.node.verify_key, peer.id)
+        if existed_peer.is_err():
+            return SyftError(
+                message=f"Failed to query peer from stash: {existed_peer.err()}"
+            )
```

Comment:
We use python>=3.10 now so can safely do this go-like thing

```suggestion
        if (err := existed_peer.is_err()):
            return SyftError(
                message=f"Failed to query peer from stash: {err}"
            )
```

but I'm not sure about including the error/exception here. @shubham3121?